### PR TITLE
suitfu: Allow to skip DFU initialization

### PR DESCRIPTION
**Pull request by tomchy**
Add Kconfig to skip SUIT DFU module initialization on system startup.

Ref: NCSDK-NONE

### CODE REVIEW

**Review by adsz-nordic — APPROVED** [review]

File: subsys/mgmt/suitfu/Kconfig [hunk 1]
```diff
@@ -16,6 +16,12 @@ config MGMT_SUITFU
 	select MCUMGR_SMP_CBOR_MIN_ENCODING_LEVEL_3
```
Comment (nordicjm) [inline subsys/mgmt/suitfu/Kconfig:18]: double newline -> reduce to 1

File: subsys/mgmt/suitfu/Kconfig [hunk 2]
```diff
@@ -16,6 +16,12 @@ config MGMT_SUITFU
...
 
 
+config MGMT_SUITFU_INITIALIZE_SUIT
+	bool "Initialize the SUIT DFU library"
+	default y
```
Comment (nordicjm) [inline subsys/mgmt/suitfu/Kconfig:21]: there is no if in this file to ensure the suitfu Kconfig is selected to gate these options

Comment (tomchy) [reply subsys/mgmt/suitfu/Kconfig:20]: That's strange.. 
Merged `menu` and `config` into `menuconfig`.

File: subsys/mgmt/suitfu/Kconfig [hunk 3]
```diff
@@ -4,17 +4,22 @@
...
 	depends on SSF_SUIT_SERVICE_ENABLED
 	depends on FLASH
 	depends on MCUMGR_SMP_LEGACY_RC_BEHAVIOUR
 	select MCUMGR_SMP_CBOR_MIN_DECODING_LEVEL_5
 	select MCUMGR_SMP_CBOR_MIN_ENCODING_LEVEL_3
 
+if MGMT_SUITFU
+
+config MGMT_SUITFU_INITIALIZE_SUIT
+	bool "Initialize the SUIT DFU library"
```
Comment (ahasztag) [inline subsys/mgmt/suitfu/Kconfig:19]: "Initialize the SUIT DFU library on startup"?

Could you add a "help" section to explain more what this actually does and that if this is not set the user must initialize the library manually?

**Review by ahasztag — COMMENTED** [review]
Seems ok when it comes to the code itself, however I'd extend the Kconfig description a bit

**Review by ahasztag — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by robertstypa — APPROVED** [review]

**Pull request merged**